engine/classes/ElggEntity.php

```diff
@@ -1068,6 +1068,47 @@ public function canEdit($user_guid = 0) {
 	}
 
 	/**
+	 * Can a user delete this entity?
+	 *
+	 * @tip Can be overridden by registering for the permissions_check:delete plugin hook.
+	 *
+	 * @param int $user_guid The user GUID, optionally (default: logged in user)
+	 *
+	 * @return bool Whether this entity is deletable by the given user.
+	 * @since 1.11
+	 * @see elgg_set_ignore_access()
+	 */
+	public function canDelete($user_guid = 0) {
+		$user_guid = (int) $user_guid;
+
+		if (!$user_guid) {
+			$user_guid = _elgg_services()->session->getLoggedInUserGuid();
+		}
+
+		// need to ignore access and show hidden entities for potential hidden/disabled users
+		$ia = elgg_set_ignore_access(true);
+		$show_hidden = access_show_hidden_entities(true);
+		
+		$user = _elgg_services()->entityTable->get($user_guid, 'user');
```

Comment:
metadata is loaded when it is first accessed; in case of checking another user his rights, this will probably be loaded in the hook. But should that not be handled in the hook?